modules/waypoint_tracking.py

```diff
@@ -23,30 +24,30 @@ def get_current_waypoint_info(
         (True, destination waypoint information), where information is (index, location).
         location can be None.
     """
-    # Download the mission commands from the drone
-    drone.commands.download()
-    drone.commands.wait_ready()
 
     # Get the current waypoint sequence
-    current_waypoint = drone.commands.next
+    current_waypoint = controller.download_commands()[1]
     waypoint_info = (current_waypoint, None)
 
     # Get the current destination
-    if current_waypoint < drone.commands.count:
-        current_command = drone.commands[current_waypoint]
-        if current_command.command == dronekit.mavutil.mavlink.MAV_CMD_NAV_WAYPOINT:
+    if current_waypoint < controller.commands.count:
+        current_command = controller.commands[current_waypoint]
+        retreive_info_success, destination_reached = (
```

Comment:
retreive_info_success rename as success? its less information but thats what the other module uses. i think it would be good to keep as a standard